nx_parallel/algorithms/centrality/betweenness.py

```diff
@@ -0,0 +1,119 @@
+      Higher values give better approximation.
+
+    normalized : bool, optional
+      If True the betweenness values are normalized by `2/((n-1)(n-2))`
+      for graphs, and `1/((n-1)(n-2))` for directed graphs where `n`
+      is the number of nodes in G.
+
+    weight : None or string, optional (default=None)
+      If None, all edge weights are considered equal.
+      Otherwise holds the name of the edge attribute used as weight.
+      Weights are used to calculate weighted shortest paths, so they are
+      interpreted as distances.
+
+    endpoints : bool, optional
+      If True include the endpoints in the shortest path counts.
+
+    seed : integer, random_state, or None (default)
+        Indicator of random number generation state.
+        See :ref:`Randomness<randomness>`.
+        Note that this is only used if k is not None.
+
+    Returns
+    -------
+    nodes : dictionary
+       Dictionary of nodes with betweenness centrality as the value.
+
+    Notes
+    -----
+    This algorithm is a parallelized version of betwenness centrality in NetworkX. Nodes are divided into
+    chunks based on the number of available processors, and otherwise all calculations are similar
+
+    References
+    ----------
+    .. [1] Ulrik Brandes:
+       A Faster Algorithm for Betweenness Centrality.
+       Journal of Mathematical Sociology 25(2):163-177, 2001.
+       https://doi.org/10.1080/0022250X.2001.9990249
+    .. [2] Linton C. Freeman:
+       A set of measures of centrality based on betweenness.
+       Sociometry 40: 35–41, 1977
+       https://doi.org/10.2307/3033543
+    .. [3] Linton C. Freeman:
+       A set of measures of centrality based on betweenness.
+       Sociometry 40: 35–41, 1977
+       https://doi.org/10.2307/3033543
+    .. [4] NetworkX Development Team:
+       Parallel Betweenness Centrality. NetworkX documentation.
+       Available at: https://networkx.org/documentation/stable/auto_examples/algorithms/plot_parallel_betweenness.html
+       Accessed on June 26, 2023.
+    """
+    # TODO: Work on passing all tests for betweenness_centrality
+    total_cores = multiprocessing.cpu_count()
+    num_nodes = len(G)
+    num_chunks = max(num_nodes // total_cores, 1)
+    node_chunks = list(_chunks(G.nodes(), num_chunks))
+    bt_sc = Parallel(n_jobs=total_cores, backend="loky")(
+        delayed(betweenness_centrality_subset)(
+            G,
+            nodes,
+            list(G),
+            normalized,
+            weight,
+        )
+        for nodes in node_chunks
+    )
+
+    #Reducing partial solution
+    bt_c = bt_sc[0]
+    for bt in bt_sc[1:]:
+        for n in bt:
+            bt_c[n] += bt[n]
+
+    return bt_c
+
+# helpers for betweenness centrality
+
+def _chunks(l, n): # divide vertices into chunks of specified size
+    l_c = iter(l)
+    while True:
+        x = tuple(itertools.islice(l_c, n))
+        if not x:
+            return
+        yield x
```

Comment:
This function is actually going to be added to `itertools` in Python 3.12 and named `batched`.  It currently appears in the [itertools docs ](https://docs.python.org/3/library/itertools.html) section at the bottom called "recipes".  Take a look and see if you prefer that code over this version. I'm sure either work.  The next "recipe" in the list after "batched" is called "grouper" and is actually much faster than "batched" but is perhaps harder to read. It is essentially `zip(*([iter(l)] * n))` in our case.  Fun with itertools. :}